python/pyarrow/util.py

```diff
@@ -78,3 +80,46 @@ def _stringify_path(path):
             return str(path)
 
     raise TypeError("not a path-like object")
+
+
+def product(seq):
+    """
+    Return a product of sequence items.
+    """
+    return functools.reduce(lambda a, b: a*b, seq, 1)
+
+
+def get_contiguous_span(shape, strides, itemsize):
+    """
+    Return a contiguous span of N-D array data.
+
+    Parameters
+    ----------
+    shape : tuple
+    strides : tuple
+    itemsize : int
+      Specify array shape data
+
+    Returns
+    -------
+    start, end : int
+      The span end points.
+    """
+    if not strides:
+        start = 0
+        end = itemsize * product(shape)
+    else:
+        start = 0
+        end = itemsize
+        for i, dim in enumerate(shape):
+            if dim == 0:
+                start = end = 0
+                break
+            stride = strides[i]
+            if stride > 0:
+                end += stride * (dim - 1)
+            elif stride < 0:
+                start += stride * (dim - 1)
+        if end - start != itemsize * product(shape):
```

Comment:
Note that this check is theoretically insufficient ;-)
```python
>>> a = np.arange(6, dtype='i1')                                                                                                                     
>>> a.shape                                                                                                                                          
(6,)
>>> a.strides                                                                                                                                        
(1,)
>>> b = as_strided(a, shape=(2, 3), strides=(5, 0))                                                                                                  
>>> b                                                                                                                                                
array([[0, 0, 0],
       [5, 5, 5]], dtype=int8)
>>> b.shape                                                                                                                                          
(2, 3)
>>> b.strides                                                                                                                                        
(5, 0)
>>> get_contiguous_span(b.shape, b.strides, b.itemsize)                                                                                              
(0, 6)   # Yet clearly b is not contiguous!
```

But I think it's good enough for now.

Reply:
`b = as_strided(a, shape=(2, 3), strides=(5, 0))` might be something that users would never do nor see :)